src/services/reducers/addReducer.js

```diff
@@ -1,18 +1,25 @@
+import { v4 as uuidv4 } from "uuid";
+import { formatDate } from "Helpers/helpers";
 
-import { ADD } from "../constants/constants";
+const initialTodos = {
+  todos: [],
+};
 
-const initialTodos = { todos: [] }
-
-const addReducer = (state=initialTodos, action) => {
-  switch (action.type){
-    case ADD:
+const addReducer = (state = initialTodos, action) => {
+  switch (action.type) {
+    case "ADD":
+      const newTodo = {
+        id: uuidv4(),
+        text: action.payload.text,
+        createdAt: formatDate(),
```

Comment:
are we creating reducer by maintaining [redux principles](https://redux.js.org/understanding/thinking-in-redux/three-principles#:~:text=Reducers%20are%20just%20pure%20functions,of%20mutating%20the%20previous%20state.)

Reply:
made changes, I think it is maintaining the principles